### DISCUSSION TITLE
Code works but some maths functions are marked red with warnings "Unresolved reference"

### DISCUSSION THREAD

**Discussion by yusenguo** [firedrakeproject/firedrake#2976] [Firedrake support] [answered]
Hi,

Please see the atteched screenshot. This happened after I updated firedrake and python (to 3.10.11). My system is macOS Ventura13.1.

The only import I use is 'from firedrake import * ' .

Would you please help me fix this?

![screenshot1](https://github.com/firedrakeproject/firedrake/assets/59740750/e1fd880f-fe64-4727-bc0b-4082222b8b2a)

**Comment by wence-** [comment]
Is your editor set up to use the firedrake virtual environment?

**Reply by yusenguo** [reply]
Yes I also checked that, and firedrake works just fine. It's just these warnings are a bit confusing.

**Reply by wence-** [reply]
It's possible that pyright (it looks like you are using vscode?) can't find the UFL functions through firedrake's "star" import.

**Reply by yusenguo** [reply]
Hi I just realised my reply was misplaced (in below) and I am copying it here in case you didn’t see - "I think you are right that the 'star import' might be the problem because when I try importing firedrake as fd and using fd.dot() etc. doesn't give a warning.  However, the screenshot is Pycharm and I also tried in VScode where the problem still exists. Any idea on why this is the case? ".

Thanks.

**Reply by dham** [reply, marked as answer]
The errors are from a static analysis tool that the IDE (Pycharm or VSCode) is running. It's not uncommon for star imports to cause this issue. It's essentially a Python limitation.

**Reply by yusenguo** [reply]
Thank you very much. I cleared caches and restarted the IDE. The errors disappeared!

**Comment by yusenguo** [comment]
I think you are right that the 'star import' might be the problem because when I try importing firedrake as fd and using fd.dot() etc. doesn't give a warning.  However, the screenshot is Pycharm and I also tried in VScode where the problem still exists.

Any idea on why this is the case?



________________________________
